src/main/java/core/basesyntax/stategy/BalanceHandler.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.stategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class BalanceHandler implements FruitOperationHandler {
+    @Override
+    public void executeOperation(FruitTransaction fruitTransaction,
+                                 Map<String, Integer> inventory) {
+        String fruit = fruitTransaction.getFruit();
+        int currentQuantity = inventory.getOrDefault(fruit, 0);
+        int newQuantity = currentQuantity + fruitTransaction.getQuantity();
+
+        if (newQuantity < 0) {
+            newQuantity = 0;
+        }
```

Comment:
Consider logging a warning or error if the transaction would result in a negative quantity. This can help identify potential issues with the transaction data.